### PR TITLE
Webcrawler

### LINKED ISSUE #3: Setup Web-Crawler for daily updates
**Describe the Issue:**
The player records in the data folder are outdated and static. Thus, they may not be enough to accurately predict player performances in current matches. Previous records were created from web-scraped data from **howstat.com.**

**Solution:**
Keep the records up to date using web-scraping for daily updates and reflect those changes in the data folder.

Comment if you would like to work on this

### PR DESCRIPTION
**Pull request by scientes**
resolves #3 

Its not completely finished, there still is one bug which i'm not able to fix which lies within the crawling framework and i can't figure out why its not working. but that part i disabled for now and only needed if we need the Long name of the player and the country he is playing for.(For people who know scrapy: Every request.meta  Dictionary entry i make vanishes when reaching the parse function of the request which for now stops me from passing the short name of the player to the parse function which determines the long name and country he's playing for.)

But for the rest here's an example output:

U G Dowe.csv
```csv
matchid,date
Matches/MatchScorecard.asp?MatchCode=0714,1973-02-16
Matches/MatchScorecard.asp?MatchCode=0693,1972-02-16
Matches/MatchScorecard.asp?MatchCode=0684,1971-04-13
Matches/MatchScorecard.asp?MatchCode=0683,1971-04-01
```

### CODE REVIEW

**Comment by lucasace** [comment]
Heyy, I would like to test this, to get a general idea of how it works, are there any major bugs in this or just minor ones ?

**Comment by scientes** [comment]
Minor Bugs which sections are disabled. So everything working runs. Word of warning IT takes about 5k-6k seconds to complete

**Comment by lucasace** [comment]
Heyy, just one thing the crawler wastes a lot of times going through matches from 1850's - 1990's . When you actually think about it, nobody from that era plays anymore, is it possible to add some kind of filter to it?

**Comment by lucasace** [comment]
It looks good, the only issue would be the full names + country as it is not being able to find them

**Comment by scientes** [comment]
Sorry i'm atm very short on time

**Comment by lucasace** [comment]
@scientes No problem! Do it at your own pace, since this is a enhancement, Its no problem if it takes some time

**Comment by lucasace** [comment]
@scientes as requested by you, I have shifted everything to a single csv in the branch `feature-singlecsv`
Coming to Long names i would actuall prefer them cause there a lot of players with the same intiials and it might cause errors . Similar for countries

**Comment by lucasace** [comment]
@scientes Heyy it would be great if you could give me a update on when u will resume
Thanks

**Comment by scientes** [comment]
Latest in a Week or so. Am 02.11.20, 06:57 schrieb Royston E Tauro <notifications@github.com>:

   
  @scientes Heyy it would be great if you could give me a update on when u will resume Thanks

**Comment by scientes** [comment]
My current solution is not the best, im atm not filtering the old games out, but i found a way to get the playernames thing working.

**Comment by lucasace** [comment]
> My current solution is not the best, im atm not filtering the old games out, but i found a way to get the playernames thing working.

That's great! You can push those changes to your branch. Ill test it out. We can make the web-crawler a WIP project if needed since you seem to be short on time.
I'll create a new branch for this. And I'll merge this there. You can continue to help once you are free.
Thank you for this! Really appreciate your help . This really is a important part for our project😄

**Comment by scientes** [comment]
My current status:
i get the player data in one file(id,name,gametype,retired)
i get the matchid in another file(grouped in folders by ODI/T20/TEST)

Missing:

- [ ] bowling

- [ ] batting

- [ ] wicketkeeping

- [ ] keeping track over different runs so that we can crawl incrementaly

- [ ] a small script via pandas which maybe cleans and reformats the data

**Comment by lucasace** [comment]
> My current status:
> i get the player data in one file(id,name,gametype,retired)
> i get the matchid in another file(grouped in folders by ODI/T20/TEST)
> 
> Missing:
> 
> * [ ]  bowling
> * [ ]  batting
> * [ ]  wicketkeeping
> * [ ]  keeping track over different runs so that we can crawl incrementaly
> * [ ]  a small script via pandas which maybe cleans and reformats the data

I believe this is good enough. Once this is merged I'll make new issues for some of them and start working on them.
The only conflicting files seems to be the `.gitignore` file should be easy to resolve
I recommend you [sync](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/syncing-a-fork) your` webcrawler` to the repo's `webcrawler` branch and then we can merge 🎉

**Comment by lucasace** [comment]
Also @scientes , even though I merged #20 , I didnt see your name in the contributors list for some reason
Just wanted to let you know, don't want it to happen in this case too

**Comment by scientes** [comment]
i've put some sample data in the output folders. not complete data from crawl

File: data_crawler/T20/match_ids.csv [hunk 1]
```diff
@@ -0,0 +1,7525 @@
...
+5009,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5003,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5526,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5527,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5007,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5009,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5011,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5013,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5529,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
+5530,Matches/MatchScorecard_T20.asp?MatchCode=0964,2019-10-18
```
Comment (lucasace) [inline data_crawler/T20/match_ids.csv:3382]: I might be wrong, does the `name` column contain `player_ids`?

**Comment by lucasace** [comment]
> i've put some sample data in the output folders. not complete data from crawl

Looks good, probably will need to make a DB server in the future
I'll be merging this soon then

File: data_crawler/T20/match_ids.csv (hunk 1 above)
Comment (scientes) [reply data_crawler/T20/match_ids.csv:3382]: oh yea

**Comment by lucasace** [comment]
@scientes Wonderful job! Thanks for contributing!! 😄 🎉

File: data_crawler/T20/match_ids.csv (hunk 1 above)
Comment (scientes) [reply data_crawler/T20/match_ids.csv:3382]: should if fix that?

**Comment by scientes** [comment]
i think the problem with #20 is that you need to contribute a certain amount of code to the default branch to be counted, i think 4 sloc or so is the minimum

File: data_crawler/T20/match_ids.csv (hunk 1 above)
Comment (lucasace) [reply data_crawler/T20/match_ids.csv:3382]: No Not neccessary as of now, Until we filter old matches, Let be there

**Pull request merged**